hybrid/compartment_examples/inter_comp_call/secure/build/Makefile.inter_comp_call

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2020-2021 The CapableVMs "CHERI Examples" Contributors.
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+# Common Makefile for the shared_object example.
+# This should not be invoked directly.
+
+ifndef BINDIR
+$(error build/Makefile.vars should set BINDIR)
+endif
+
+ifdef RUNDIR
+RUNDIR := $(RUNDIR)/inter_comp_call
+else
+RUNDIR := inter_comp_call
+endif
+
+UPDATE_DDC_EXAMPLE := secure-update_ddc
+
+EXAMPLES = secure secure-redirect_clr secure-try_deref secure-update_ddc
+CFILES := $(foreach dir, $(EXAMPLES), $(wildcard $(dir)/*.c))
+SHARED_SOURCES := $(wildcard shared/*.S shared/*.s)
+HFILES := $(wildcard include/*.h ../include/*.h)
+
+.PHONY: all clang-format clean run-inter_comp_call
+
+all: $(BINDIR)/main
+
+clang-format:
+	$(CFORMAT) -i $(CFILES) $(HFILES)
+
+clean:
+	if [ -d bin ]; then \
+		rm -rf $(BINDIR); \
+		rmdir bin --ignore-fail-on-non-empty; \
+	fi
+
+.SECONDEXPANSION:
+$(BINDIR)/%: $$(wildcard %/*.s) $$(wildcard %/*.S) $$(wildcard %/*.c) $(SHARED_SOURCES)
+	@mkdir -p $(BINDIR)
+	$(CC) $(CFLAGS) -L $(BINDIR) $+ -o $@
+
+# switch_compartment.S conditionally includes one of the two clean
+# implementations. Setting INTER_COMP_LEAK_X20 tells switch-compartment.S to
+# select the "broken" version of clean (which simulates leaking a register).
+# In the UPDATE_DDC_EXAMPLE example, x20 contains the DDC of the compartment
+# switcher.
+run-inter_comp_call-$(UPDATE_DDC_EXAMPLE): CFLAGS += -D INTER_COMP_LEAK_X20
```

Comment:
Let me try to understand: in `UPDATE_DDC_EXAMPLE`, we have a list of examples for which we want to run with the "leaky" version of `clean`. This basically says "if you do `make run-inter_comp_call-{example}`, if that `{example}` is one of `UPDATE_DDC_EXAMPLE`, we will be adding the "leaky flag"". Did I get that right?

Reply:
Yep you got it right, except `UPDATE_DDC_EXAMPLE` can't be a list (this assumes `UPDATE_DDC_EXAMPLE` is a single example).